### PR TITLE
Bug 1219302 - Support removing files using the preprocessor when a flag is enabled r=rchien

### PR DESCRIPTION
**Pull request by samgiles**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e1500c2c2736c10028cddceac8f1e8bfc0b771e4

File: build/preprocessor.js [hunk 1]
```diff
@@ -112,18 +112,24 @@ function processFiles(flag, enable, list, stagePath) {
...
 
-exports.execute = function(options, flag, list) {
+exports.execute = function(options, flag, list, removeFilesIfEnabled) {
   var stagePath = options.STAGE_APP_DIR;
   var enable = options[flag] === '1';
 
   processFiles(flag, enable, list.process, stagePath);
 
-  if (!enable && list.remove) {
+  var shouldRemoveFiles = (removeFilesIfEnabled || !enabled) && list.remove;
```
Comment (rickychien) [inline build/preprocessor.js]: ```enabled``` -> ```enable```

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b463322f8aa9bb15a2c7cd959b96acf453fbeff0

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=74b4e62cf1408ab1205c89740e59756d4f83f191

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=5c6e15b317869a2172adae1ae70798d67ade2c7d

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=bccc6e6cbc25eee3e8299277565451adfa1b74fd

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=d85ec65421cc5163ae85c6a4f94fe4022142b0f2

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f2eeb0b932032cddc9eb75fcf2fb159a91ae07da

**Comment by autolander** [comment]
The pull request could not be applied to the integration branch. Please try again after current integration is complete. You may need to rebase your branch against the target branch.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=04177bb60d0040a5fc5b414e71b89fff61c46172

**Comment by autolander** [comment]
The pull request could not be applied to the integration branch. Please try again after current integration is complete. You may need to rebase your branch against the target branch.

**Pull request merged**